go/libraries/doltcore/sqle/dfunctions/hashof.go

```diff
@@ -82,6 +82,9 @@ func (t *HashOf) Eval(ctx *sql.Context, row sql.Row) (interface{}, error) {
 		}
 
 		cm, err = ddb.ResolveRef(ctx, branchRef)
+		if err != nil {
```

Comment:
Same comment as above. If we're going to return an err from inside conditional branch blocks, do it everywhere. 

(This one is an error because of shadowing, a problem the one in the block above doesn't have. But it's not easy to tell the difference, and we should always return errors on the line after getting one returned, not wait until after the branch logic merges together)